doc/sphinx/proofs/writing-proofs/equality.rst

```diff
@@ -31,44 +31,64 @@ There are multiple notions of :gdef:`equality` in Coq:
   other advantages.  ":term:`Convertible <convertible>`" is another way of saying that
   two terms are definitionally equal.
 
-Equality
---------
-
-.. tacn:: f_equal
-
-   For a goal with the form :n:`f a__1 ... a__n = g b__1 ... b__n`, creates
-   subgoals :n:`f = g` and :n:`a__i = b__i` for the `n` arguments. Subgoals
-   that can be proven by :tacn:`reflexivity` or :tacn:`congruence` are solved
-   automatically.
+Reflexivity, symmetry and transitivity tactics
+----------------------------------------------
 
 .. tacn:: reflexivity
 
-   For a goal with the form `t=u`, verifies that `t` and `u` are :term:`convertible`,
-   and if so,
+   For a goal with the form `t = u`, verifies that `t` and `u` are
+   :term:`definitionally equal <definitional equality>`, and if so,
    solves the goal.  If not, it fails.  It is equivalent to ``apply refl_equal``.
 
-   .. exn:: The conclusion is not a substitutive equation.
-      :undocumented:
+   The tactic may also be applied to goals with the form `R @term__1 @term__2` where
+   `R` is an equivalence relation registered with the `Equivalence` or `Reflexive`
+   typeclasses.  See :cmd:`Class` and :cmd:`Instance`.
+
+   .. todo PR: an "equivalence relation" is not necessarily a "setoid equality", right?
```

Comment:
A setoid is a set (or type) equipped with an equivalence relation, and this equivalence relation should be a congruence for the operations that are defined on the elements of the set / type. To rewrite, we need the congruence properties, but for using the `reflexivity`, `symmetry` and `transitivity` tactics, we only need the equivalence properties. Actually, we only need the corresponding properties for each tactic, so we can use `reflexivity` with any (declared) reflexive relation, `symmetry` with any (declared) symmetric relation and `transitivity` with any (declared) transitive relation, even if these relations are not equivalence relations (or setoid equalities).

That being said, the same is true for `rewrite`: we don't strictly need a setoid equality. Sometimes, we can rewrite with weaker relations as long as the declared properties are sufficient for the specific use case.